pkg/tfshim/sdk-v2/instance_state.go

```diff
@@ -81,7 +88,7 @@ func objectFromCtyValue(v cty.Value) map[string]interface{} {
 	contract.AssertNoErrorf(err, "Failed to marshal cty.Value to a JSON string value")
 
 	var m map[string]interface{}
-	err = json.Unmarshal(buf.Bytes(), &m)
+	err = unmarshalJSON(buf.Bytes(), &m)
```

Comment:
Yeah this is interesting, we were trying to economize on code but losing some control in the process. Looks like `marshal` right above this line is inlined from TF SDK to customize it. With some further elbow grease we could fuse marshal with unmarshalJSON into one function, that is have marshal emit the desired map[string]interface{} without turning around through JSON. Perhaps worth a comment in the source but not blocking on it.